test-d/partial-on-undefined-deep.ts

```diff
@@ -0,0 +1,46 @@
+import {expectType} from 'tsd';
+import type {PartialOnUndefinedDeep} from '../index';
+
+declare const foo: PartialOnUndefinedDeep<{
+  func: (() => void) | undefined;
+  obj: {objKey: 1} | undefined;
+  objDeep: {
+    subObj: string | undefined;
+  };
+  str: string | undefined;
+  union: 'test1' | 'test2' | undefined;
+  number: number | undefined;
+  bool: boolean | undefined;
+  date: Date | undefined;
+  regexp: RegExp | undefined;
+  symbol: Symbol | undefined;
+  null: null | undefined;
+  record: Record<string, any> | undefined;
+  map: Map<string, string> | undefined;
+  set: Set<string> | undefined;
+  array: any[] | undefined;
+  tuple: ['test1'] | undefined;
+  readonly: readonly string[] | undefined;
+}>;
+
+expectType<{
+  func?: typeof foo['func'];
+  obj?: typeof foo['obj'];
```

Comment:
Don't use abbreviations. Applies everywhere.
```suggestion
  object?: typeof foo['object'];
```